validphys2/src/validphys/overfit_metric.py

```diff
@@ -144,22 +147,28 @@ def array_expected_overfitting(
         (number_of_resamples*Npdfs,) sized array containing the mean delta chi2
         values per resampled list.
     """
-    fitted_val_erf = np.array([info.validation for info in replica_data])
-
-    number_pdfs = calculate_chi2s_per_replica.shape[0]
-    list_expected_overfitting = []
-    for _ in range(number_pdfs * number_of_resamples):
-        mask = np.random.randint(
-            0, number_pdfs, size=int(resampling_fraction * number_pdfs)
-        )
-        res_tmp = calculate_chi2s_per_replica[mask][:, mask]
-
-        fitted_val_erf_tmp = fitted_val_erf[mask]
-        expected_val_chi2 = res_tmp.mean(axis=0)
-        delta_chi2 = fitted_val_erf_tmp - expected_val_chi2
-        expected_delta_chi2 = delta_chi2.mean()
-
-        list_expected_overfitting.append(expected_delta_chi2)
+    # calculate_chi2s_per_replica is set to zero if the pseudodata generation 
+    # has changed sinc the fit has been performed. As a result the overfitting
+    # metric can no longer be determined.
+    if (calculate_chi2s_per_replica == 0).all():
+        list_expected_overfitting = calculate_chi2s_per_replica
```

Comment:
Maybe it is better than in the case in which it cannot be computed the table is populated with "N/A" instead of [0, 0, NaN] which might confuse people.